### ISSUE TITLE
How can I ship an icon (or font) in a fragment in a MSIX?

### ISSUE THREAD

**Issue by davidanthoff** [microsoft/terminal#10359]
I am now shipping a profile fragment in the Julia MSIX Windows Store installer [here](https://www.microsoft.com/store/apps/9NJNWW8PVKMN). It works great! I still have one question, though: how can I ship an icon in my fragment that will be used? I tried putting an icon file next to the fragment json and then just adding `"icon": "julia.ico"` to the fragment, but with that it complained that it couldn't find the icon.

I think I need some way to specify a relative path here that is relative to the fragment file location, right? Is there some way to do that?

**Comment by skyline75489** [comment]
The exact code is [this one](https://github.com/WhitewaterFoundry/Pengwin/blob/master/Pengwin/Public/Fragments/pengwin.json), I think.

**Comment by davidanthoff** [comment]
Hm, interesting... That does seem to work, but it also seems a bit hacky? What they are doing is copying the icon file at startup of the program into the local app data location (which I think means it will only work once everything has run once), and then they have semi hard coded the path to where MSIX is storing local app data in the fragment for the icon path. That latter step is probably not a good idea? Isn't that file location considered an implementation detail of MSIX? Essentially they are using a special path to reach into the container of their app, which seems not in the spirit of MSIX.

Is there maybe a way to do the same thing in a less involved way? Couldn't the terminal just be able to interpret relative paths from the fragment?

**Comment by skyline75489** [comment]
Yeah .. I wouldn't say "semi hard coded" path is a good idea or an elegant solution, either. I'll leave this to @DHowett and @PankajBhojwani

**Comment by WSLUser** [comment]
You know this also raises the question for me of whether you can import a fragment yourself as a user. Say you import the rootfs of Pengwin and install it through the `wsl --import` option. You won't be able to obtain the fragment through the Store as a result of this method. Perhaps I should raise an issue for this if there isn't one already. This would also affect Julia and any other third party application, especially those that aren't available in the Store.

**Comment by mpela81** [comment]
I guess #6747 would be helpful (but probably not easy to implement), you could embed the icon into the json itself without worrying about file paths

**Comment by DHowett** [comment]
This is an excellent question. I somewhat wonder if icon resource paths like `ms-appx://My.PackageName_myPublisher/Public/XYZ` work...

If they do: cool, the first free thing we've gotten from the app platform.

If not: I think we will want to add manual support for something like it.

**Comment by davidanthoff** [comment]
I'm trying that now!

**Comment by davidanthoff** [comment]
@DHowett I tried this now, but it does not seem to work. My fragment now is [this](https://github.com/JuliaLang/juliaup/blob/master/msix/Fragments/Julia.json), and that version ships in the Windows Store as the "Julia" app, so you should be able to try that and see where it doesn't work.

**Comment by DHowett** [comment]
Thanks!

**Comment by zadjii-msft** [comment]
Tentatively sticking on backlog, though we may want to bump this up to a 2.0 priority.
